common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationAwareUTF8String.java

```diff
@@ -415,6 +415,23 @@ else if (codePoint == 0x03C2) {
     }
   }
 
+  private static final int COMBINED_LOWERCASE_I_DOT = 0x69 << 16 | 0x307;
```

Comment:
this should not be a valid code point (I'll check manually to confirm this)

Reply:
also, TODO: add comment explaining that this is for internal use only, and is in fact not a valid code point